dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -57,23 +53,88 @@ public CultureInfo Culture
     /// </summary>
     public ILoggerFactory LoggerFactory { get; }
 
+    /// <summary>
+    /// Kernel context reference
+    /// </summary>
+    public IKernel Kernel => this.GetKernelContext();
+
+    /// <summary>
+    /// Spawns the kernel for the context.
+    /// </summary>
+    /// <remarks>
+    /// The kernel context is a lightweight instance of the main kernel with its services.
+    /// </remarks>
+    /// <returns>Kernel reference</returns>
+    private IKernel GetKernelContext()
+        => this._originalKernel; // TODO: Clone a lightweight kernel instead of returning the same instance
+
     /// <summary>
     /// Constructor for the context.
     /// </summary>
+    /// <param name="kernel">Kernel reference</param>
     /// <param name="variables">Context variables to include in context.</param>
     /// <param name="skills">Skills to include in context.</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
     public SKContext(
+        IKernel kernel,
         ContextVariables? variables = null,
         IReadOnlySkillCollection? skills = null,
         ILoggerFactory? loggerFactory = null)
     {
+        Verify.NotNull(kernel, nameof(kernel));
+
+        // 
+        this._originalKernel = kernel;
         this.Variables = variables ?? new();
         this.Skills = skills ?? NullReadOnlySkillCollection.Instance;
         this.LoggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
         this._culture = CultureInfo.CurrentCulture;
     }
 
+    /// <summary>
```

Comment:
I plan to hide the public constructor (future change) so that only a Kernel creates an instance of SKContext -- a couple practical considerations to sort out first. 

These extra constructors shouldn't be needed (the first one covers all cases).